src/components/FiltersPopover/FiltersPopover.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useRef } from 'react';
+import styles from './FiltersPopover.module.css';
+import Button from '../Button';
+import ButtonGroup from '../ButtonGroup';
+import Icon from '../Icon';
+import Popover, { PopoverProps } from '../Popover';
+
+export type FiltersPopoverProps = {
+  /**
+   * Form controls, form fields, or other relevant information that will be displayed in the filters popover.
+   */
+  children: React.ReactNode;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * CSS class names that can be appended to the footer button group.
+   */
+  footerButtonGroupClassName?: string;
+  /**
+   * Indicates status that filters have been selected, influencing toggle button variant.
+   */
+  hasSelectedFilters?: boolean;
+  /**
+   * Callback called when the clear button is called.
+   */
+  onClear?: () => void;
+  /**
+   * Callback called when filters drawer is closed.
+   */
+  onClose?: () => void;
+  /**
+   * Callback called when the apply button is called.
+   */
+  onApply?: () => void;
+  /**
+   * Popover placement options relative to the filters trigger button.
+   */
+  placement?: PopoverProps['placement'];
+  /**
+   * Text to be placed in the button that activates the Filters Popover
+   */
+  triggerText: string;
+};
+
+type FiltersPopoverRenderProps = {
+  /**
+   * Closes popover action render prop passed from Popover parent.
+   */
+  close: (
+    focusableElement?: HTMLElement | React.RefObject<HTMLElement>,
+  ) => void;
+  /**
+   * Popover open status render prop passed from Popover parent.
+   */
+  open: boolean;
+} & Omit<FiltersPopoverProps, 'placement'>;
+
+/**
+ * This helper component passes render props from <Popover> to handle onClose functionality
+ * that HeadlessUI Popover does not provide natively.
+ */
+const FiltersPopoverRender = ({
+  children,
+  className,
+  close,
+  footerButtonGroupClassName,
+  hasSelectedFilters,
+  onApply,
+  onClear,
+  onClose,
+  open,
+  triggerText,
+  ...other
+}: FiltersPopoverRenderProps) => {
+  /**
+   * Hooks to emulate an onClose callback for the Popover.
+   */
+  const firstRender = useRef(true);
+  useEffect(() => {
+    if (firstRender.current) {
+      firstRender.current = false;
+    } else if (!open && onClose) {
+      onClose();
+    }
+  }, [open]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const buttonVariant = hasSelectedFilters ? 'primary' : 'secondary';
+  const buttonStatus = hasSelectedFilters ? 'brand' : 'neutral';
+
+  const buttonGroupClassName = clsx(
+    styles['footer__button-group'],
+    footerButtonGroupClassName,
+  );
+
+  return (
+    <>
+      <Popover.Button as={React.Fragment}>
+        <Button
+          className={styles['filters-popover__button']}
+          status={buttonStatus}
+          variant={buttonVariant}
+        >
+          <Icon name="filter-list" purpose="decorative" size="1.5rem" />
+          {triggerText}
```

Comment:
Could also have `triggerText` also replace Icon as well and have default trigger text as the icon + 'Filters'

Reply:
I think having a standardized icon here makes sense because the action is always the same (like a next button always has a right arrow regardless of whether the text says "Next" or "Continue").